src/program/downloaders/alldebrid.py

```diff
@@ -51,594 +34,185 @@ def validate(self) -> bool:
-                            params[f"magnets[{i}]"] = magnet
-    
-                        response = get(f"{AD_BASE_URL}/magnet/instant", params=params, additional_headers=self.auth_headers, proxies=self.proxy, response_type=dict, specific_rate_limiter=self.inner_rate_limit, overall_rate_limiter=self.overall_rate_limiter)
-                        if response.is_ok and self._evaluate_stream_response(response.data, processed_stream_hashes, item, stream_hashes):
-                            return True
-                    except Exception as e:
-                        logger.error(f"Error checking cache for streams: {str(e)}", exc_info=True)
-    
-        logger.log("NOT_FOUND", f"No wanted cached streams found for {item.log_string} after processing all chunks")
-        return False
-
-    def _evaluate_stream_response(self, data, processed_stream_hashes, item, stream_hashes):
-        """Evaluate the response data from the stream availability check."""
-        if data.get("status") != "success":
-            logger.error("Failed to get a successful response")
-            return False
-    
-        magnets = data.get("data", {}).get("magnets", [])
-        for magnet in magnets:
-            stream_hash = magnet.get("hash")
-            stream_hash_lower = stream_hash.lower() if stream_hash else None
-    
-            # Skip if the stream has already been processed or if the hash is not valid
-            if not stream_hash_lower or stream_hash_lower in processed_stream_hashes:
-                continue
-    
-            # Mark the stream as processed
-            processed_stream_hashes.add(stream_hash_lower)
-    
-            if not magnet.get("instant", False):
+    def process_hashes(
+        self, chunk: list[str], needed_media: dict, break_pointer: list[bool]
+    ) -> dict:
+        return self.get_cached_containers(chunk, needed_media, break_pointer)
+
+    def download_cached(self, active_stream: dict) -> str:
+        torrent_id = add_torrent(active_stream.get("infohash"))
+        if torrent_id:
+            self.existing_hashes.append(active_stream.get("infohash"))
+            return torrent_id
+        raise Exception("Failed to download torrent.")
+
+    def get_cached_containers(
+        self, infohashes: list[str], needed_media: dict, break_pointer: list[bool]
+    ) -> dict:
+        """
+        Get containers that are available in the debrid cache containing `needed_media`
+
+        Parameters:
+        - infohashes: a list of hashes that might contain the data we need
+        - needed_media: a dict of seasons, with lists of episodes, indicating what content is needed
+        - break_pointer: first bool indicates if the needed content was found yet, 2nd pointer indicates if we should break once it's found.
+        """
+        cached_containers = {}
+        response = get_instant_availability(infohashes)
+        magnets = {m.get("hash"): m for m in response}
+
+        for infohash in infohashes:
```

Comment:
This whole block was super-confusing, and I'm still not convinced I got it right.

The RealDebrid API seems to require you to pass the individual files you want added, which makes the API more complicated.
AllDebrid is all-or-nothing for a magnet so I tried to mimic how RD behaves by just passing all the files. I'm still confused by what a "container" is though because the RD API docs are pretty terrible.

It seems like a better abstraction here might help?